spacy/lookups.py

```diff
@@ -192,7 +192,7 @@ def __init__(self, name=None, data=None):
         self.name = name
         # Assume a default size of 1M items
         self.default_size = 1e6
```

Comment:
I'm not entirely sure how the performance degrades if the size is way too small, but for the lookups tables we have, the average size is 250K and the median size is 5K. Maybe something like 10K is a better default? (@matt, what do you think?)

Reply:
10k seems good to me. Would that be a minimum or a just a default if no `data` is provided? I'm thinking about the empty tables in `lookups_extra.bin`—if 10k isn't a minimum, then the Bloom filters will still have a size of 1, which will render them useless if the user adds to the tables.

But, if we make 10k a minimum, does that mean that we should initialize the size to `len(data) + 10000` when `data` is provided? That is, if we assume the tables aren't read-only, then should we allocate extra space? If we don't, the Bloom filter will still work, but the error rate will be higher. I'm not sure if we need to account for that or not.

Reply:
That's a good point. Matt knows more about the performance details and I'm still not sure what he thinks about the default size here. Hopefully if performance is crucial people will look into the details a bit here like you have, but it is pretty buried in the code.

The main use case I could imagine where the extra lookups tables grow is that someone loads a provided model with an empty prob table and wants to add their own probabilities, which could easily be for 1M+ entries. But the memory usage for the much more common case where the empty probs table isn't touched should be reasonable and should be the main priority, I think.

Reply:
Maybe `size = max(len(data), self.default_size)` with a default size of 10k?